qubesadmin/tools/qvm_check.py

```diff
@@ -20,15 +20,33 @@
 # with this program; if not, write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
-""" Exits sucessfull if the provided domains exists, else returns failure """
+""" Exits sucessfull if the provided domain(s) exist, else returns failure """
 
 import sys
 
 import qubesadmin.tools
 import qubesadmin.vm
 
-parser = qubesadmin.tools.QubesArgumentParser(description=__doc__,
-                                              vmname_nargs='+')
+class QvmCheckArgumentParser(qubesadmin.tools.QubesArgumentParser):
+    """Collecting error message(s) on invalid domain(s) instead of aborting"""
+    def __init__(self, description, vmname_nargs):
+        super().__init__(description=description, vmname_nargs=vmname_nargs)
+        self._invalid_domains = []
+
+    def error(self, message):
+        if message.startswith('no such domain: '):
+            self._invalid_domains.append(message[17:-1])
```

Comment:
I don't like this approach for two reasons:
- matching error message by string is fragile
- `error` method normally terminates the program, changing this behavior is asking for troubles

Better extend QubesArgumentParser with this functionality (an option to collect invalid domains instead of exiting?), instead of catching the error after the fact.

Reply:
> Better extend QubesArgumentParser with this functionality (an option to collect invalid domains instead of exiting?), instead of catching the error after the fact.

It could be done. But the patch will be longer. As two classes have to be extended. I will work on it.

Reply:
It is done. I am waiting for CI/CD bot to re-evaluate it to see if any additional unit test is necessary.